cpp/ql/src/semmle/code/cpp/Declaration.qll

```diff
@@ -124,7 +124,7 @@ class Declaration extends Locatable, @declaration {
    * To test whether this declaration has a particular name in the global
    * namespace, use `hasGlobalName`.
    */
-  abstract string getName();
+  string getName() { none() }
```

Comment:
```suggestion
  string getName() { none() } // overridden in subclasses
```

Please include this comment, also in the pseudo-abstract predicates below. I think it's helpful both for casual readers and for our future selves.

Reply:
Doh! I completely forgot about this comment. I've included it in all the places I've removed an abstract keyword from a predicate now.